documentation/design/console/global-console.adoc

```diff
@@ -0,0 +1,134 @@
+:toc:
+:toclevels: 4
+
+== Overview
+
+The Console is EnMasse comprises two separate web-applications that are hyperlinked together.
+
+* Global Console - that allows the viewing and management of address spaces.
+* Address Space Console - that allows the viewing and management of:
+ * addresses within an address space, and
+ * messaging connections made by applications within that address space.
+ * statistics
+
+In the long term, the desire is that current responsibilities of the Address Space Console will be migrated into the Global Console and the separate Address Space Consoles will be retired completed.
+
+In addition it is desired that the Global Console is extensible to manage other Custom Resources too. The use-case here is to manage the Custom Resources that correspond to other messaging infrastructure for instance Apache Strimzi. Th architecture will allow the view and the control of the Custom Resource to be plugged in. This will allow the UI to present a multi-stage create workflow targeted to the Resource in question.
+
+In the long term, the Global Console will also present key statistics about each managed resource. This will allow the user to understand a glance the state of the whole system. The statistics being presented will vary depending on the resource in question. Statistic won't be included in the first phase.
+
+== Global Console Design
+
+The Global Console is a Pattern Fly 4/React-JS application.  It is served by Oauth-proxy (https://github.com/openshift/oauth-proxy or https://github.com/pusher/oauth2_proxy).
+
+Authentication is delegated to the Oauth-proxy component.
+
+- For OpenShift, OpenShift Oauth-Proxy is configured to authenticate against the OAuth2 server of OpenShift itself.
+- For Kubernetes, Pusher Oauth-Proxy is configued with Open ID Connect (OIDC).  This needs to be the same OIDC instance that Kubernetes is configured to use itself.
+
+The UI will use the Kubernetes API to list and control the resources that it needs to expose for Management. This part of the UI will be extensible so that the ability to control other resources can be enabled. In the first the phase the two resource types exposed for management will be EnMasse's standard addressspace and brokered addressspace.
+
+Initially the UI will periodically poll the API for the list of resources (standard address space and brokered address space). Later this will evolve to utilise watch for a more responsive UI with fewer overheads.
```

Comment:
We will also need a console server in order to move to a watch.  
Currently in openshift, we are fetching the namespaces that the user has permissions to.  Then fetching the addressspaces for each namespace.  This would require a watch for each project + 1 for the project list.  Chrome has a limit of 30 websockets per host.
It would be best to use one websocket, which watches a console server.

Reply:
Agreed.